src/celrender/vertexobject.cpp

```diff
@@ -160,9 +154,110 @@ void VertexObject::disableAttribArrays() const noexcept
         glDisableVertexAttribArray(p.location);
 }
 
+void
+VertexObject::disableVertexAttribArray(GLint location) const noexcept
+{
+    glDisableVertexAttribArray(location);
+}
+
+void
+VertexObject::setVertexAttribConstant(GLint location, float value) const noexcept
+{
+    disableVertexAttribArray(location);
+    glVertexAttrib1f(location, value);
+}
+
+void
+VertexObject::enableVertexAttribArray(GLint location) const noexcept
+{
+    glEnableVertexAttribArray(location);
+}
+
 void VertexObject::setVertexAttribArray(GLint location, GLint count, GLenum type, bool normalized, GLsizei stride, GLsizeiptr offset)
 {
     assert(location >= 0);
     m_attribParams.emplace_back(location, offset, stride, count, type, normalized);
 }
+
+IndexedVertexObject::IndexedVertexObject(GLenum indexType) :
+    VertexObject(),
+    m_indexType(indexType)
+{
+}
+
+IndexedVertexObject::IndexedVertexObject(GLsizeiptr bufferSize, GLenum streamType, GLenum indexType, GLsizeiptr indexSize) :
+    VertexObject(bufferSize, streamType),
+    m_indexType(indexType),
+    m_indexStreamType(streamType),
+    m_indexSize(indexSize)
+{
+}
+
+IndexedVertexObject::IndexedVertexObject(GLsizeiptr bufferSize, GLenum streamType, GLenum indexType, GLsizeiptr indexSize, GLenum indexStreamType) :
+    VertexObject(bufferSize, streamType),
+    m_indexType(indexType),
+    m_indexStreamType(indexStreamType),
+    m_indexSize(indexSize)
+{
+}
+
+IndexedVertexObject::~IndexedVertexObject()
+{
+    if (m_vioId != 0)
+        glDeleteBuffers(1, &m_vioId);
+}
+
+void
+IndexedVertexObject::bindWritable() noexcept
+{
+    m_state |= State::Update;
+    bind();
+}
+
+void
+IndexedVertexObject::bind() noexcept
+{
+    if (getStreamType() != GL_STATIC_DRAW)
+        VertexObject::bind();
+
+    if ((m_state & State::Initialize) != 0)
+        glGenBuffers(1, &m_vioId);
+
+    if (!isVAOSupported() ||
```

Comment:
I think the condition here is wrong - as far as I can tell the buffer never gets bound which is why I'm seeing the segfaults. Verified by changing from `!isVAOSupported()` to `isVAOSupported()` which stops the crashes.

Reply:
or index buffer is not bound to vao actually

Reply:
Please check now

Reply:
No, GL_ELEMENT_ARRAY_BUFFER should be managed by Vertex Array Objects

Reply:
Please check now

Reply:
Segfault no longer occurs, thanks!